### PR TITLE
[ID-29][ID-22][ID-33]Feat circular menu canvas

### PR DESCRIPTION
**Pull request by DongNguyenQ**

### CODE REVIEW

File: lib/features/home/widgets/google_map.dart [hunk 1]
```diff
@@ -8,6 +8,11 @@ import 'package:google_maps_flutter/google_maps_flutter.dart';
 import 'package:interactive_diary/constants/map_style.dart';
 import 'package:interactive_diary/gen/assets.gen.dart';
 
+const String menuCameraMarkerLocationId = 'menuCameraMarkerLocation';
+const String menuPencilMarkerLocationId = 'menuPencilMarkerLocation';
+const String menuEmojiMarkerLocationId = 'menuEmojiMarkerLocation';
+const String menuVoiceMarkerLocationId = 'menuVoiceMarkerLocationId';
```
Comment (suesitran) [inline lib/features/home/widgets/google_map.dart:14]: Shouldn't all the ID strings end with Id?

File: lib/features/home/widgets/google_map.dart [hunk 2]
```diff
@@ -172,18 +193,166 @@ class _GoogleMapViewState extends State<GoogleMapView>
...
     if (pngBytes != null) {
       if (!_streamController.isClosed) {
-        _streamController.sink.add(Uint8List.view(pngBytes.buffer));
+        markers.add(Marker(
+            markerId: const MarkerId('currentLocation'),
+            position: widget.currentLocation,
+            icon: BitmapDescriptor.fromBytes(Uint8List.view(pngBytes.buffer)),
+            zIndex: 1,
+            onTap: () {
+              print('BASE MARKER CLICKED');
```
Comment (suesitran) [inline lib/features/home/widgets/google_map.dart]: Please avoid print(), you can use debugPrint() instead

File: lib/features/home/widgets/google_map.dart [hunk 3]
```diff
@@ -172,18 +193,166 @@ class _GoogleMapViewState extends State<GoogleMapView>
...
+    popupVoiceAnimation = _declareMenuIconsAnimation(
+        start: baseAnchor, end: Offset(xVoice, yVoice), controller: controller);
+
+    currentLocationAnimation = Tween<double>(begin: 0, end: 1)
+        .animate(CurvedAnimation(parent: controller, curve: Curves.elasticOut));
+  }
+
+  void _generateCircularMenuIcons() async {
+    markers.removeWhere(
+            (Marker element) => element.markerId.value != 'currentLocation');
```
Comment (suesitran) [inline lib/features/home/widgets/google_map.dart]: I think this currentLocation ID can be defined as constant as well

File: lib/features/home/widgets/google_map.dart [hunk 4]
```diff
@@ -172,18 +193,166 @@ class _GoogleMapViewState extends State<GoogleMapView>
...
+    if (_controller.status != AnimationStatus.dismissed) {
+      markers.addAll(<Marker>{
+        Marker(
+            markerId: const MarkerId(menuCameraMarkerLocationId),
+            position: widget.currentLocation,
+            icon: cameraMarkerBitmap,
+            anchor: popupCameraAnimation.value,
+            onTap: () {
+              print('menuCameraMarkerLocation');
```
Comment (suesitran) [inline lib/features/home/widgets/google_map.dart]: Avoid using print

File: lib/features/home/widgets/google_map.dart [hunk 5]
```diff
@@ -172,18 +193,166 @@ class _GoogleMapViewState extends State<GoogleMapView>
...
+            }),
+        Marker(
+            markerId: const MarkerId(menuPencilMarkerLocationId),
+            position: widget.currentLocation,
+            icon: penMarkerBitmap,
+            anchor: popupPenAnimation.value,
+            onTap: () {
+              print('menuPencilMarkerLocation');
```
Comment (suesitran) [inline lib/features/home/widgets/google_map.dart]: Same

File: lib/features/home/widgets/google_map.dart [hunk 6]
```diff
@@ -172,18 +193,166 @@ class _GoogleMapViewState extends State<GoogleMapView>
...
+            }),
+        Marker(
+            markerId: const MarkerId(menuEmojiMarkerLocationId),
+            position: widget.currentLocation,
+            icon: emojiMarkerBitmap,
+            anchor: popupEmojiAnimation.value,
+            onTap: () {
+              print('menuEmojiMarkerLocation');
```
Comment (suesitran) [inline lib/features/home/widgets/google_map.dart]: Same

File: lib/features/home/widgets/google_map.dart [hunk 7]
```diff
@@ -172,18 +193,166 @@ class _GoogleMapViewState extends State<GoogleMapView>
...
+            }),
+        Marker(
+            markerId: const MarkerId(menuVoiceMarkerLocationId),
+            position: widget.currentLocation,
+            icon: voiceMarkerBitmap,
+            anchor: popupVoiceAnimation.value,
+            onTap: () {
+              print('menuVoiceMarkerLocation');
```
Comment (suesitran) [inline lib/features/home/widgets/google_map.dart]: Same

File: lib/generated/intl/messages_all.dart [hunk 8]
```diff
@@ -0,0 +1,62 @@
+// DO NOT EDIT. This is code generated via package:intl/generate_localized.dart
+// This is a library that looks up messages for specific locales by
+// delegating to the appropriate library.
+
+// Ignore issues from commonly used lints in this file.
```
Comment (suesitran) [inline lib/generated/intl/messages_all.dart]: Something is not right with this file. There should be many strings in here. 

You can choose to NOT commit this file, or commit the file with all strings in it.

File: lib/generated/intl/messages_en.dart [hunk 9]
```diff
@@ -0,0 +1,25 @@
+// DO NOT EDIT. This is code generated via package:intl/generate_localized.dart
+// This is a library that provides messages for a en locale. All the
+// messages from the main program should be duplicated here with the same
+// function name.
+
+// Ignore issues from commonly used lints in this file.
+// ignore_for_file:unnecessary_brace_in_string_interps, unnecessary_new
```
Comment (suesitran) [inline lib/generated/intl/messages_en.dart]: Same as above

File: packages/nartus_ui_package/ios/Flutter/Generated.xcconfig [hunk 10]
```diff
@@ -0,0 +1,13 @@
+// This is a generated file; do not edit or check into version control.
```
Comment (suesitran) [inline packages/nartus_ui_package/ios/Flutter/Generated.xcconfig]: Please do not commit this file to repo

File: packages/nartus_ui_package/ios/Flutter/flutter_export_environment.sh [hunk 11]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+# This is a generated file; do not edit or check into version control.
```
Comment (suesitran) [inline packages/nartus_ui_package/ios/Flutter/flutter_export_environment.sh]: Please do not commit this file to repo

File: packages/nartus_ui_package/macos/Flutter/ephemeral/Flutter-Generated.xcconfig [hunk 12]
```diff
@@ -0,0 +1,11 @@
+// This is a generated file; do not edit or check into version control.
```
Comment (suesitran) [inline packages/nartus_ui_package/macos/Flutter/ephemeral/Flutter-Generated.xcconfig]: Please do not commit this file to repo

File: packages/nartus_ui_package/macos/Flutter/ephemeral/flutter_export_environment.sh [hunk 13]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+# This is a generated file; do not edit or check into version control.
```
Comment (suesitran) [inline packages/nartus_ui_package/macos/Flutter/ephemeral/flutter_export_environment.sh]: Please do not commit this file to repo

**Review by suesitran — CHANGES_REQUESTED** [review]
Please remove those unexpected files

File: packages/nartus_ui_package/ios/Flutter/Generated.xcconfig (hunk 10 above)
Comment (DongNguyenQ) [reply packages/nartus_ui_package/ios/Flutter/Generated.xcconfig]: @suesitran it's weird that i didn't commit those files, i think it was committed when i pull new update from repo

**Review by suesitran — APPROVED** [review]

**Pull request merged**